xmake/core/package/package.lua

```diff
@@ -2268,27 +2269,46 @@ end
 function _instance:_generate_build_configs(configs, opt)
     opt = opt or {}
     configs = table.join(self:fetch_librarydeps(), configs)
-    if self:is_plat("windows") then
-        local ld = self:build_getenv("ld")
-        local runtimes = self:runtimes()
-        -- since we are ignoring the runtimes of the headeronly library,
-        -- we can only get the runtimes from the dependency library to detect the link.
-        if self:is_headeronly() and not runtimes and self:librarydeps() then
-            for _, dep in ipairs(self:librarydeps()) do
-                if dep:is_plat("windows") and dep:runtimes() then
-                    runtimes = dep:runtimes()
-                    break
-                end
+    -- since we are ignoring the runtimes of the headeronly library,
+    -- we can only get the runtimes from the dependency library to detect the link.
+    local runtimes = self:config("runtimes") or self:runtimes()
```

Comment:
self:runtimes() will get runtimes from `self:config("runtimes")`

we need only ..

```lua
local runtimes = self:runtimes()
```